lib/propolis/src/hw/nvme/admin.rs

```diff
@@ -110,13 +117,19 @@ impl NvmeCtrl {
 
         // Finally, create the Submission Queue
         match self.create_sq(
-            cmd.qid,
+            super::queue::CreateParams {
+                id: cmd.qid,
+                base: GuestAddr(cmd.prp),
+                size: cmd.qsize,
+            },
+            false,
             cmd.cqid,
-            GuestAddr(cmd.prp),
-            cmd.qsize,
-            mem,
+            nvme,
         ) {
-            Ok(_) => cmds::Completion::success(),
+            Ok(sq) => {
+                self.io_sq_post_create(nvme, sq);
```

Comment:
i'm not sure if this is exactly how `cpuid_boot_test_migration` ended up panicking. but if the device is reset after `create_sq` returns Ok and before `io_sq_post_create` has gone and associated the queues, then a reset could witness that `NvmeCtrl::sqs[i]` has a queue while there's not actually a minder for the queue yet.

it... seems like it'd be better to forbid the inconsistent state between these "control plane" operations on the disk, than to tolerate the inconsistency because it might be a very rare race. an "administrative" `Mutex<()>` we use to serialize queue creation/deletion/device reset? i haven't fully thought through this, what do you think?

(i'm sad i'd thought about evil devices and did not think about concurrent operations...)